department-of-veterans-affairs/caseflow
# Intake | More refactoring to support two intake paths

More refactoring to support existing features of intake on both intake paths.

- Add Review page that renders differently depending on the form
- Make cancel an endpoint on `IntakesController` instead of `RampElectionIntakes` controller. Allow all types of intakes to be cancelled via that endpoint
- Add the complete suite of `RampElection` specs into the feature flag path. The old tests will be deleted after the feature flag is turned on.
- Add automatic re-routing to the formSelection page (this was required to get the old specs to work)

## To test
- Complete a Ramp election intake start to finish under the old method
- Try to visit the form selection page while an intake was started. It should redirect you to the review page.
- Start a Ramp refiling intake. Cancel should work.

## Notes
- This PR is a little bigger than average since there was a lot of structural changes. A lot of the added code will be removed upon feature completion
